Assets/MixedRealityToolkit/_Core/Utilities/FadeColors.cs

```diff
@@ -0,0 +1,275 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Utilities
+{
+    /// <summary>
+    /// Fades colors (alpha value) on a material or TextMesh, requires a transparent material
+    /// </summary>
+    public class FadeColors : MonoBehaviour
+    {
+        /// <summary>
+        /// The selectable ease types for the animation
+        ///     Linear: steady progress
+        ///     EaseIn: ramp up in speed
+        ///     EaseOut: ramp down in speed
+        ///     EaseInOut: ramp up then down in speed
+        /// </summary>
+        public enum LerpTypes { Linear, EaseIn, EaseOut, EaseInOut }
+
+        [Tooltip("Fade ease type")]
+        public GameObject TargetObject;
+
+        [Tooltip("Fade ease type")]
+        public LerpTypes LerpType;
+
+        [Tooltip("Time to fade in seconds")]
+        public float LerpTime = 1f;
+
+        [Tooltip("Run by default? or Status")]
+        public bool IsRunning = false;
+
+        [Tooltip("Animation is complete!")]
+        public UnityEvent OnComplete;
+
+        /// <summary>
+        /// Return the current alpha value
+        /// </summary>
+        public float GetCurrentAlpha { get { return mCurrentAlpha; } }
+
+        // animation ticker
+        private float mLerpTimeCounter;
+
+        // color and alpha values
+        private Color mCachedColor;
+        private float mCurrentAlpha;
+
+        // component targets
+        private Material mMaterial;
+        private TextMesh mTextMesh;
+
+        // are we fading in or out?
+        private bool mIsFadingIn = true;
```

Comment:
Let's also update the names of these to comply with our usual standard. (drop the m prefixes)